phoenixRest/models/core/agenda_entry.py

```diff
@@ -30,22 +30,52 @@ class AgendaEntry(Base):
     event_uuid = Column(UUID(as_uuid=True), ForeignKey("event.uuid"), nullable=False)
     event = relationship("Event")
     
+    title = Column(Text, nullable=False)
+    description = Column(Text, nullable=True, server_default=None)
+    location = Column(Text, nullable=True, server_default=None)
     time = Column(DateTime, nullable=False)
 
-    title = Column(Text, nullable=False)
-    description = Column(Text, nullable=False)
+    deviating_time = Column(DateTime, nullable=True, server_default=None)
+    deviating_location = Column(Text, nullable=True, server_default=None)
+    deviating_information = Column(Text, nullable=True, server_default=None)
+
+    state_pinned = Column(Boolean, server_default='false')
+    state_deviating_time_unknown = Column(Boolean, server_default='false')
+    state_cancelled = Column(Boolean, server_default='false')
+
+    log_created_by_uuid = Column(UUID(as_uuid=True), ForeignKey("user.uuid"), nullable=False)
+    log_created_by_time = Column(DateTime, nullable=False, server_default='NOW()')
+    log_modified_by_uuid = Column(UUID(as_uuid=True), ForeignKey("user.uuid"), nullable=True, server_default=None)
+    log_modified_by_time = Column(DateTime, nullable=True, server_default=None)
 
-    def __init__(self, title: str, description: str, time: datetime, event: Event):
+    def __init__(self, event: Event, title: str, description: str, location: str, time: DateTime, state_pinned: bool, log_created_by_uuid: str):
+        self.event = event
         self.title = title
         self.description = description
-        self.event = event
+        self.location = location
         self.time = time
+        self.state_pinned = state_pinned
+        self.log_created_by_uuid = log_created_by_uuid
+        self.log_created_by_time = datetime.now()
```

Comment:
Better off as a server default. Not that I do that either.

Reply:
Consider this: all fields with an intuitive default value should be populated by sql, so it is easy to insert as raw SQL where possible. Even though we should never do it, it's a good way to think.